splunk_add_on_ucc_framework/generators/globalConfig_generator/create_minimal_globalconfig.py

```diff
@@ -0,0 +1,74 @@
+#
+# Copyright 2025 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from splunk_add_on_ucc_framework.generators.globalConfig_generator import (
+    GlobalConfigGenerator,
+)
+from typing import Any, Dict, Union
+import os
+import addonfactory_splunk_conf_parser_lib as conf_parser
+
+
+class MinimalGlobalConfig(GlobalConfigGenerator):
```

Comment:
Hm, would it make sense for this logic to be part of an existing `globalConfig`? This breaks the whole integration with `...Generator` classes but it would be consistent with the existing class.

Reply:
The idea was, since we are generating globalConfig, I kept it under `...Generator` class. Moreover, if we move it to GlobalConfig class of `global_config.py`, we would be updating the behaviour of that class - currently, it is to parse a globalConfig and convert it to a dictionary and provide properties to access various attributes. Moving this implementation there, this class would be responsible to create a physical globalConfig file (if not present) and then continue with the current behaviour. Additionally, we would require to give the `source_dir` path (which induces additional changes in the tests).

Reply:
Migrated the implementation from `Filegenerator` to `GlobalConfig` class.